content/docs/intro/concepts/programming-model.md

```diff
@@ -706,93 +1348,106 @@ url := virtualmachine.DnsName().Apply(func(dnsName string) (interface{}, error)
 
 ```javascript
-let certCertificate = new aws.acm.Certificate("cert", {
-  domainName: "example.com",
-  validationMethod: "DNS",
-});
-let certValidation = new aws.route53.Record("cert_validation", {
-  // Need to pass along a deep subproperty of this Output
-  records: [certCertificate.domainValidationOptions.apply(domainValidationOptions => domainValidationOptions[0].resourceRecordValue)],
-  ...
+var pulumi = require("@pulumi/pulumi");
+// ...
+let connectionString = pulumi.all([sqlServer.name, database.name])
+    .apply(([server, db]) => `Server=tcp:${server}.database.windows.net;initial catalog=${db}...`);
 ```
 
 ```typescript
-let certCertificate = new aws.acm.Certificate("cert", {
-  domainName: "example.com",
-  validationMethod: "DNS",
-});
-let certValidation = new aws.route53.Record("cert_validation", {
-  // Need to pass along a deep subproperty of this Output
-  records: [certCertificate.domainValidationOptions.apply(domainValidationOptions => domainValidationOptions[0].resourceRecordValue)],
-  ...
+import * as pulumi from "@pulumi/pulumi";
+// ...
+let connectionString = pulumi.all([sqlServer.name, database.name])
+    .apply(([server, db]) => `Server=tcp:${server}.database.windows.net;initial catalog=${db}...`);
 ```
 
 ```python
-certificate = aws.acm.Certificate("cert",
-  domain_name="example.com",
-  validation_method="DNS",
-
-record = aws.route53.Record("validation",
-  # Need to pass along a deep subproperty of this Output
-  records=[certificate.domain_validation_options.apply(
-      lambda domain_validation_options: domain_validation_options[0].resource_record_value
-  )],
-  ...
+from pulumi import Output
+# ...
+connection_string = Output.all(sql_server.name, database.name) \
+    .apply(lambda args: f"Server=tcp:{args[0]}.database.windows.net;initial catalog={args[1]}...")
 ```
 
 ```go
-// Helpers for accessing properties are not yet available in Go.
+// `all` is not yet available in Go.
 //
 // See https://github.com/pulumi/pulumi/issues/1614.
 ```
 
 ```csharp
-// Helpers for accessing properties are not yet available in .NET.
+// In .NET 'Output.Tuple' is used so that each unwrapped value will preserve their distinct type.
+// 'Output.All' can be used when all input values have the same type (i.e. all are Output<string>)
+var connectionString = Output.Tuple(sqlServer.name, database.name)
+    .Apply(t => `Server=tcp:${t.Item1}.database.windows.net;initial catalog=${t.Item2}...`);
 ```
 
-To make this kind of property and array element access more simple, an `Output` _lifts_ the properties of the value that is wrapped, allowing you to access them directly off of the `Output` itself.  This allows the previous example to be written more simply as:
+Notice that {{< pulumi-all >}} works by itself returning an output that represents the combination of all of the input outputs, so that within the callback, the raw values are available inside of [a tuple](https://en.wikipedia.org/wiki/Tuple).
```

Comment:
`input outputs` reads a bit oddly -- maybe something like this?

```suggestion
Notice that {{< pulumi-all >}} works by itself returning an output that represents the combination of multiple outputs, so that within the callback, the raw values are available inside of [a tuple](https://en.wikipedia.org/wiki/Tuple).
```